### DISCUSSION TITLE
Redirects and Authentication Timing

### DISCUSSION THREAD

**Discussion by rkunboxed** [csells/go_router#112] [Q&A] [answered]
Question on redirects for auth vs. unauth users. Currently, I trigger a session check on `initState` of the app. I need GoRouter to hold off on any redirects until the session check is complete. What is the best way to go about this? The GoRouter redirects are handled before my session check has completed so it treats everyone as unauthenticated when they first enter the app. Subsequent redirects work great because the session check has completed.

**Comment by csells** [comment, marked as answer]
The way I do this is to initialize my auth provider in `main` before calling `runApp`. However, you can also do this with a refreshListenable that starts at a startup screen before the logged in state is known.

**Reply by rkunboxed** [reply]
I ended up putting my auth check in `main` before calling `runApp` as you suggested so it would run as early as possible and it worked like a charm. Thanks!

**Comment by csells** [comment]
You could also take advantage of [the new navigator builder feature](https://pub.dev/packages/go_router#navigator-builder) to show a splash screen before even showing the navigator, although you'll still need to use that in conjunction with a `refreshListenable` once your login sequence has completed.

**Reply by rkunboxed** [reply]
Thanks, I will give this approach a try.

**Comment by csells** [comment]
excellent news
